### PR TITLE
Feature/15006 service manager check windows

### PR DESCRIPTION
**Pull request by pris54**

### CODE REVIEW

File: admin/assets/stylesheets/elements/_sorting.scss [hunk 1]
```diff
@@ -0,0 +1,13 @@
+.sort {
+  display: inline-block;
+  width: 0;
+  height: 0;
+  border-style: solid;
+  border-width: 12px 6px 0 6px;
```
Comment (jon-shipley) [inline admin/assets/stylesheets/elements/_sorting.scss]: How does this 12px border look on mobile?  Should we use `per()` to convert automatically to em?

File: admin/controllers/administrator.js [hunk 2]
```diff
@@ -115,8 +117,87 @@ const setUpdateTiming = async (req, res, next) => {
...
+    if (startYear === endYear) {
+      startYear = ''
+    }
+    return startDate.format('DD MMM') + startYear + ' to ' + endDate.format('DD MMM YYYY')
+  }
+
+  try {
+    checkWindows = await CheckWindow.getCheckWindows(sortField, sortDirection)
+    checkWindowsFormatted = checkWindows.map(cw => {
+      const adminStartDateMo = moment(cw.adminStartDate)
```
Comment (jon-shipley) [inline admin/controllers/administrator.js]: We haven't used the postfix notation here anywhere else that I know of.  I'd say drop the `Mo` bit.

File: admin/models/check-window.js [hunk 3]
```diff
@@ -4,24 +4,50 @@ const mongoose = require('mongoose')
...
-    type: [{ type: Number, ref: 'CheckForm', required: true }]
+    type: [{
+      type: Number,
+      ref: 'CheckForm',
+      required: true
+    }]
   }
 }, {timestamps: true})
 
+CheckWindow.statics.getCheckWindows = function (sortBy, direction) {
```
Comment (jon-shipley) [inline admin/models/check-window.js]: Some jsdoc would be good

File: admin/test/features/assign_check_to_windows.feature [hunk 4]
```diff
@@ -46,11 +46,13 @@ Feature:
     Given I am on the assign check window page
     Then check windows should have a end date
 
+  @wip
```
Comment (jon-shipley) [inline admin/test/features/assign_check_to_windows.feature]: @activemq  Shouldn't this be remove before merging?  Are these tests complete?

File: pupil-spa/src/app/check-complete/check-complete.component.spec.ts [hunk 5]
```diff
@@ -1,14 +1,21 @@
 import { async, ComponentFixture, TestBed } from '@angular/core/testing';
 
 import { CheckCompleteComponent } from './check-complete.component';
+import { AuditService } from '../services/audit/audit.service';
```
Comment (jon-shipley) [inline pupil-spa/src/app/check-complete/check-complete.component.spec.ts]: Why is there pupil-spa code in the PR?

File: admin/test/features/assign_check_to_windows.feature (hunk 4 above)
Comment (activemq) [reply admin/test/features/assign_check_to_windows.feature]: @jon-shipley No this is basically these tests were working fine before this ticket was picked up. The BA's seemed to forgot to add AC's around assigning checks to a window, so now this functionality needs to added as a separate ticket....Thus the tests are marked @wip

File: pupil-spa/src/app/check/check.component.ts [hunk 6]
```diff
@@ -135,16 +130,12 @@ export class CheckComponent implements OnInit {
...
         const matches = /^Q(\d+)$/.exec(stateDesc);
-        this.registerInputService.flush();
         this.question = this.questionService.getQuestion(parseInt(matches[ 1 ], 10));
-        this.registerInputService.initialise();
         this.viewState = 'question';
         break;
       }
       case(/^complete$/).test(stateDesc):
         // Show the check complete screen
-        this.registerInputService.flush();
```
Comment (GeorgeChatzigiannis) [inline pupil-spa/src/app/check/check.component.ts]: Why did this get removed?
Seems like a bad conflict resolution

**Comment by pris54** [comment]
New PR #167 covers this work.

**Pull request closed without merging**